dev-tools/test.sh

```diff
@@ -25,38 +26,51 @@ while test $# -gt 0; do
     esac
 done
 
-if [[ $CHANGED -gt 0 ]]; then
-    # if --changed flag was passed: only run changed tests and don't update dependency database
-    TESTMON="--testmon-nocollect"
-    CHANGED_MESSAGE=" affected by recent changes"
+# The default pytests args we use
+PYTEST_ARGS=("--quiet" "--disable-warnings" "--numprocesses=auto")
 
-    # check if .testmondata file exists; if not, inform that all tests will be run
-    if [ ! -f ".testmondata" ]; then
+# Check if --changed flag was passed
+if [[ -n "${CHANGED}" ]]; then
+    # Check if .testmondata file exists
+    if [[ -f ".testmondata" ]]; then
+        # Only run changed tests and don't update dependency database
+        PYTEST_ARGS+=("--testmon-nocollect")
+        CHANGED_MESSAGE=" affected by recent changes"
+    else
+        # Inform that all tests will be run
         echo -e "\nIt looks like you have not run pytest without the \"--changed\" flag before." | print_warning
         echo -e "Pytest has to build a dependency database by running all tests without the flag once.\n" | print_warning
+        # Override test path argument
         unset TEST_PATH
-        TESTMON="--testmon"
-        unset CHANGED_MESSAGE
```

Comment:
Instead of always setting the message and unsetting it in the if case, I directly converted it to an if/else